lib/pbench/server/api/resources/query_apis/query_dataset_list.py

```diff
@@ -0,0 +1,290 @@
+        uri_fragment = gen_month_range(self.prefix, ".v6.run-data.", start, end)
+
+        uri = f"{self.es_url}/{uri_fragment}/_search"
+        try:
+            # query Elasticsearch
+            es_response = requests.post(
+                uri,
+                params={"ignore_unavailable": "true"},
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+                json=payload,
+            )
+            es_response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            self.logger.exception("HTTP error {} from Elasticsearch post request", e)
+            abort(502, message="INTERNAL ERROR")
+        except requests.exceptions.ConnectionError:
+            self.logger.exception(
+                "Connection refused during the Elasticsearch post request"
+            )
+            abort(502, message="Network problem, could not post to Elasticsearch")
+        except requests.exceptions.Timeout:
+            self.logger.exception(
+                "Connection timed out during the Elasticsearch post request"
+            )
+            abort(504, message="Connection timed out, could not post to Elasticsearch")
+        except requests.exceptions.InvalidURL:
+            self.logger.exception(
+                "Invalid url {} during the Elasticsearch post request", uri
+            )
+            abort(500, message="INTERNAL ERROR")
+        except Exception as e:
+            self.logger.exception(
+                "Exception {!r} occurred during the Elasticsearch post request", e
+            )
+            abort(500, message="INTERNAL ERROR")
+        else:
+            datasets = []
+            try:
+                es_json = es_response.json()
+                hits = es_json["hits"]["hits"]
+                self.logger.info("{} controllers found", len(hits))
+                for dataset in hits:
+                    src = dataset["_source"]
+                    run = src["run"]
+                    d = {
+                        "key": run["name"],
+                        "run.name": run["name"],
+                        "run.controller": run["controller"],
+                        "run.start": run["start"],
+                        "run.end": run["end"],
+                        "id": run["id"],
+                    }
+                    try:
+                        timestamp = parser.parse(run["start"]).utcfromtimestamp()
+                    except Exception as e:
+                        self.logger.info(
+                            "Can't parse start time {} to integer timestamp: {}",
+                            run["start"],
+                            e,
+                        )
+                        # fall back to the end timestamp sort key (why?)
+                        # TODO: Should this just abort instead?
+                        timestamp = dataset["sort"][0]
```

Comment:
I imagine we've selected the sort key as a fallback because we know it's "good"...if it weren't then ES would have rejected the document at ingestion...right??

However, if the start time is bad, why should we "abort"?  (And, what would that mean?...drop this data set, fail the whole request, or what?)